src/config.rs

```diff
@@ -0,0 +1,178 @@
+pub struct Config {
+    /// Configuration for the `format` command.
+    pub format: FormatConfig,
+    /// Configuration for the `check` and `lint` commands.
+    pub check: CheckConfig,
+    /// Configuration for the `run` command.
+    pub run: RunConfig,
+    /// Common configuration options for all commands.
+    pub common: CommonConfig,
+}
+
+/// Represents shared configuration options for Sprocket commands.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct CommonConfig {
+    /// Display color output.
+    pub color: bool,
+    /// The report mode.
+    pub report_mode: Mode,
+}
+
+impl Default for CommonConfig {
+    fn default() -> Self {
+        Self {
+            color: true,
+            report_mode: Mode::default(),
+        }
+    }
+}
+
+/// Represents the configuration for the Sprocket `format` command.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct FormatConfig {
+    /// Use tabs for indentation (default is spaces).
+    pub with_tabs: bool,
+    /// The number of spaces to use for indentation levels (default is 4).
+    pub indentation_size: usize,
+    /// The maximum line length (default is 90).
+    pub max_line_length: usize,
+}
+
+impl Default for FormatConfig {
+    fn default() -> Self {
+        Self {
+            with_tabs: false,
+            indentation_size: 4,
+            max_line_length: 90,
+        }
+    }
+}
+
+/// Represents the configuration for the Sprocket `check` and `lint` commands.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct CheckConfig {
+    /// Rule IDs to except from running.
+    pub except: Vec<String>,
+    /// Causes the command to fail if any warnings are reported.
+    pub deny_warnings: bool,
+    /// Causes the command to fail if any notes are reported.
+    pub deny_notes: bool,
+    /// Hide diagnostics with `note` severity.
+    pub hide_notes: bool,
+}
+
+/// Represents the configuration for the Sprocket `run` command.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct RunConfig {
+    /// Path to the engine configuration file.
+    pub config: PathBuf,
+}
+
+impl Config {
+    /// Create a new config instance by reading potential configurations.
+    pub fn new(path: Option<String>) -> Self {
+        // Check for a config file in the current directory
+        // Start a new Figment instance with default values
+        let mut figment = Figment::new().admerge(Serialized::from(Config::default(), "default"));
+
+        // Check XDG_CONFIG_HOME for a config file
+        // On MacOS, check HOME for a config file
+        #[cfg(target_os = "macos")]
+        {
+            if let Some(home) = dirs::home_dir() {
+                tracing::info!(
```

Comment:
These are probably `trace!` statements, only in the rarest of occasions are you going to want this level of information IMO.